integration_test/testutil_test.go

```diff
@@ -37,8 +37,9 @@ import (
 const (
 	persistentIndexCache = "krew-persistent-index-cache"
 	krewBinaryEnv        = "KREW_BINARY"
-	validPlugin          = "konfig" // a plugin in central index with small size
-	validPlugin2         = "mtail"  // a plugin in central index with small size
+	validPlugin          = "konfig"           // a plugin in central index with small size
+	validPlugin2         = "mtail"            // a plugin in central index with small size
+	validNonIndexPlugin  = "konfig-non-index" // a plugin not in central index with small size
```

Comment:
instead of doing this let's do this:

1. test.WithIndex()
2. find konfig.yaml in the index path, delete it
3. use upgrade --no-update-index in both `install` and `upgrade` calls in the integration test.

this way:
- you can fake the situation that konfig is not in the index.
- without having to duplicate a testdata file (I really dislike increasing testdata)

Reply:
Sounds much better, updated.